src/Extensions/Script/Serialization/JavaScriptSerializer.cs

```diff
@@ -1,42 +1,283 @@
 // MIT License.
 
-using System.Collections.Specialized;
+using System.Collections;
+using System.Diagnostics.CodeAnalysis;
 using System.Text;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using System.Web.Resources;
 
 namespace System.Web.Script.Serialization;
 
-// TODO: We'll want to use System.Text.Json and change it to be strongly typed
-internal sealed class JavaScriptSerializer
+public class JavaScriptSerializer
 {
-    public object MaxJsonLength { get; internal set; }
+    internal const string ServerTypeFieldName = "__type";
+    internal const int DefaultRecursionLimit = 100;
+    internal const int DefaultMaxJsonLength = 2097152;
+    private static readonly JavaScriptSerializer _defaultJavaScriptSerializer = new();
 
-    internal static string SerializeInternal(OrderedDictionary orderedDictionary)
+    private readonly JsonSerializerOptions _options = new()
     {
-        throw new NotImplementedException();
+        MaxDepth = DefaultRecursionLimit,
+    };
+
+    internal static string SerializeInternal(object o)
+    {
+        return _defaultJavaScriptSerializer.Serialize(o);
+    }
+
+    internal static object Deserialize(JavaScriptSerializer serializer, string input, Type type, int depthLimit)
+    {
+        ArgumentNullException.ThrowIfNull(input);
+
+        if (input.Length > serializer.MaxJsonLength)
+        {
+            throw new ArgumentException(AtlasWeb.JSON_MaxJsonLengthExceeded, nameof(input));
+        }
+
+        return JsonSerializer.Deserialize(input, type, serializer._options);
     }
 
-    internal void Serialize(object value, StringBuilder builder, SerializationFormat javaScript)
+    // INSTANCE fields/methods
+    private readonly JavaScriptTypeResolver _typeResolver;
```

Comment:
can you consolidate the fields and remove unnecessary comments? this class is now a new implementation so we should clean it up